### DISCUSSION TITLE
Strapi : How is it possible to keep track of user who created/updated any content?

### DISCUSSION THREAD

**Discussion by vinod77** [strapi/strapi#6401] [Legacy Topics]
I want to know user who created & updated any record, but currently Strapi doesn't provides this feature or if i am wrong, please let me know how do i know about this.

Currently i can only see CreatedAt & UpdatedAt attributes which are working good.

**It will be great to have CreatedBy & UpdatedBy as well. This feature is really important for us.**

Thanks!

**Comment by lauriejim** [comment]
Hello! By default it's not possible...

BUT here is my workaround.

**WORKROUND**

The ctx.state.user of ctx.state.admin is define in a policy form the Users-Permissions plugin. So if we want to access this data, we will have to place our code after.

My first idea was to use a middleware to apply it to every request, but the variable is not define at this time.

But I remember how we force the permission policy to every routes in the Users-Permissions plugin.

[strapi/strapi](https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-users-permissions/middlewares/users-permissions/index.js)

This set the permission policy in first position for every routes of the application.

So to do what I want, I will use the exact same thing.

First we will create our policy to add our logic.

**Path —** `config/policies/ctx.js`

```jsx
module.exports = async (ctx, next) => {
  // manage raw requests
  if (ctx.request.body) {
    ctx.request.body._user = ctx.state.user;
    ctx.request.body._admin = ctx.state.admin;
  }

  // manage form data requests
  if (ctx.request.data) {
    const data = JSON.parse(ctx.request.body.data);
    data._user = ctx.state.user;
    data._admin = ctx.state.admin;
    ctx.request.body.data = JSON.stringify(data);
  }

  await next();
};
```

Here we are adding the user or admin in a temp variable in the body request.

Then let apply this policy to every routes

**Path —** `middlewares/ctx/index.js`

```jsx
const _ = require('lodash');

module.exports = strapi => {
  return {
    initialize() {
      _.forEach(strapi.admin.config.routes, value => {
        if (_.get(value.config, 'policies')) {
          value.config.policies.splice(1, 0, 'global::ctx');
        }
      });

      _.forEach(strapi.config.routes, value => {
        if (_.get(value.config, 'policies')) {
          value.config.policies.splice(1, 0, 'global::ctx');
        }
      });

      if (strapi.plugins) {
        _.forEach(strapi.plugins, plugin => {
          _.forEach(plugin.config.routes, value => {
            if (_.get(value.config, 'policies')) {
              value.config.policies.splice(1, 0, 'global::ctx');
            }
          });
        });
      }
    },
  };
};
```

We are using the same logic as the Users-Permissions plugin but we place our `ctx` policy in second position, just after the `permission` one.

After that we will have to apply our login in the life cycle function of our APIs

**Path —** `api/article/models/article.js`

```jsx
module.exports = {
  lifecycles: {
    async beforeUpdate(params, data) {
      console.log(data._admin);
    },
    async afterUpdate(result, params, data) {
      console.log(data._user);
    }
  }
};
```

**Comment by vinod77** [comment]
Great.. It works now
Thanks @lauriejim
